ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskViewModel.kt

```diff
@@ -17,21 +17,46 @@ package com.google.android.ground.ui.datacollection.tasks.location
 
 import android.content.res.Resources
 import android.location.Location
+import androidx.lifecycle.viewModelScope
 import com.google.android.ground.ui.datacollection.tasks.AbstractTaskViewModel
 import com.google.android.ground.ui.datacollection.tasks.location.CaptureLocationTaskResult.Companion.toCaptureLocationResult
 import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+/** Location lock states relevant for attempting to enable it or not. */
+enum class LocationLockState {
+  /** The default, unknown state. */
+  UNKNOWN,
+
+  /** The location lock was already enabled, or an attempt was made. */
+  ALREADY_ENABLED,
+
+  /** The location lock was not already enabled. */
+  NEEDS_ENABLE,
+
+  /** Trigger to enable the location lock. */
+  ENABLE,
+}
 
 class CaptureLocationTaskViewModel @Inject constructor(resources: Resources) :
   AbstractTaskViewModel(resources) {
 
   private val lastLocation = MutableStateFlow<CaptureLocationTaskResult?>(null)
+  /**
+   * Allows control for triggering the location lock programmatically.
+   */
+  val enableLocationLockFlow = MutableStateFlow(LocationLockState.UNKNOWN)
 
   suspend fun updateLocation(location: Location) {
     lastLocation.emit(location.toCaptureLocationResult())
   }
 
   fun updateResponse() {
-    setValue(lastLocation.value)
+    if (lastLocation.value == null) {
+      viewModelScope.launch { enableLocationLockFlow.emit(LocationLockState.ENABLE) }
+    } else {
+      setValue(lastLocation.value)
+    }
```

Comment:
Is this scenario even possible? Ideally we should disable the `Capture location` button if there is no location to be set.

Reply:
That makes sense! Thank you!